### ISSUE TITLE
using batchnorm2d, torchscript network output differently with original network when change to eval mode

### ISSUE THREAD

**Issue by liuhuaiyu** [pytorch/pytorch#14027]
## To Reproduce

I used `BatchNorm2d()` in the network (**repo is below**), and converted it to TorchScript version.
When give the same input, the output is different when changed original network to eval mode. (there's no `dropout`)
Example results are bellow (compare MSE between 2 output):

`default train vs torch_script train:`
`tensor(0., grad_fn=<PowBackward0>)`
`default eval vs torch_script eval:`
`tensor(56.4769, grad_fn=<PowBackward0>)`
`default train vs torch_script eval:`
`tensor(0., grad_fn=<PowBackward0>)`
`default eval vs torch_script train:`
`tensor(56.4769, grad_fn=<PowBackward0>)`

## Environment
You can get the script and run it with:
```
https://github.com/liuhuaiyu/PyTorch-Issue
python why_bn.py
```

 - PyTorch Version (preview):
 - OS (Ubuntu 16.04):
 - How you installed PyTorch ( `pip`):
 - Python version: 2.7

**Comment by soumith** [comment]
cc: @driazati @zdevito 

this is expected at the moment but aiming to be fixed as far as I know.

**Comment by driazati** [comment]
Fixed in #14719
